langserver/definition.go

```diff
@@ -108,14 +133,36 @@ func (h *LangHandler) handleXDefinition(ctx context.Context, conn jsonrpc2.JSONR
 		return nil, err
 	}
 
-	var nodes []*ast.Ident
+	var nodes []foundNode
 	obj, ok := pkg.Uses[node]
 	if !ok {
 		obj, ok = pkg.Defs[node]
 	}
 	if ok && obj != nil {
 		if p := obj.Pos(); p.IsValid() {
-			nodes = append(nodes, &ast.Ident{NamePos: p, Name: obj.Name()})
+			typ := pkg.TypeOf(node).String()
```

Comment:
TypeOf documents that this could return nil. Is it possible in this context? (maybe not due to us doing a type check)